app/src/main/java/org/jellyfin/androidtv/ui/playback/segments/SegmentSkipFragment.kt

```diff
@@ -68,17 +71,23 @@ class SegmentSkipFragment() : Fragment() {
 		val currentSegment = getCurrentSegment(currentPosition) ?: lastSegment ?: return
 		lastSegment = currentSegment
 
-		val shouldShowButton = buttonConfig?.skipButtonVisible == true &&
+		val shouldPerformSkip = buttonConfig?.skipButtonVisible == true &&
 			currentPosition >= currentSegment.showAt.millis &&
 			currentPosition < currentSegment.hideAt.millis
 
-		if (shouldShowButton && button.visibility != View.VISIBLE) {
+
+
+		if (shouldPerformSkip && button.visibility != View.VISIBLE && skipTypeController[UserPreferences.skipMode] == SegmentSkipType.ShowButton) {
```

Comment:
Using isVisible here instead of visibility will make the code easier to follow. 

isVisible is true for View.VISIBLE and false for View.GONE
It can be used as a getter and a setter